### PR TITLE
feat: add NestedDictInput filter and non-vector search for AstraVectorStoreComponent

### PR DESCRIPTION
**Pull request by mieslep**
This pull request introduces several enhancements and refactorings to the `AstraDB` vector store component. The most significant changes include the introduction of a new `NestedDictInput` for advanced search filters, improvements to how search arguments are built, and better error handling during the search process.

### Enhancements to search filters and arguments:

* Introduced `NestedDictInput` for `advanced_search_filter` and deprecated the existing `search_filter` (`src/backend/base/langflow/components/vectorstores/astradb.py`).
* Enhanced `_build_search_args` method to clean and validate search filters, and to log deprecated usage of `search_filter` (`src/backend/base/langflow/components/vectorstores/astradb.py`).

### Error handling and logging improvements:

* Improved error handling in `search_documents` method by adding specific error messages and logging steps (`src/backend/base/langflow/components/vectorstores/astradb.py`).

### Codebase improvements:

* Added a cached vector store attribute `_cached_vector_store` to `AstraVectorStoreComponent` (`src/backend/base/langflow/components/vectorstores/astradb.py`).
* Updated import statements to include `AstraDBVectorStore` and `NestedDictInput` (`src/backend/base/langflow/components/vectorstores/astradb.py`).

### CODE REVIEW

File: src/backend/base/langflow/components/vectorstores/astradb.py [hunk 1]
```diff
@@ -482,43 +491,69 @@ def _map_search_type(self) -> str:
...
+        if query:
+            args = {
+                "query": query,
+                "search_type": self._map_search_type(),
+                "k": self.number_of_results,
+                "score_threshold": self.search_score_threshold,
+            }
+        elif self.advanced_search_filter or search_filter:
+            args = {
+                "n": self.number_of_results,
```
Comment (erichare) [inline src/backend/base/langflow/components/vectorstores/astradb.py:508]: Should this be k?

**Review by erichare — APPROVED** [review]

**Review by erichare — CHANGES_REQUESTED** [review]
One small param issue (possibly)

File: src/backend/base/langflow/components/vectorstores/astradb.py (hunk 1 above)
Comment (mieslep) [reply src/backend/base/langflow/components/vectorstores/astradb.py:508]: No, `metadata_search` uses `n`, not `k`. 

https://python.langchain.com/api_reference/astradb/vectorstores/langchain_astradb.vectorstores.AstraDBVectorStore.html#langchain_astradb.vectorstores.AstraDBVectorStore.metadata_search

Comment (erichare) [reply src/backend/base/langflow/components/vectorstores/astradb.py:508]: Hah, you're right indeed. I'm surprised it differs but hey, thats correct :) Approving! Thanks!

**Pull request merged**